src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,39 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector }) {
+    const popupElement = document.querySelector(popupSelector);

```

Comment:
`    const popupElement = document.querySelector(popupSelector);
`

It's not needed here because you have `this._popupElement` as the popup


**No** need to search the **modal** again in the children of `Popup` because the parent has already found it in its constructor as `this._popupElement`. It’s available in any child.